lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -258,16 +282,22 @@ rollForwardUTxO isOurs block u =
     slot = Read.getEraSlotNo $ Read.getEraBHeader block
 
 rollBackward
-    :: Read.ChainPoint
+    :: ResolveSlot
+    -> Read.ChainPoint
     -> WalletState
     -> (WalletState, Read.ChainPoint)
-rollBackward targetPoint w =
+rollBackward resolveSlot targetPoint w =
     ( w
         { walletTip = actualPoint
         , utxoHistory =
             UTxOHistory.rollBackward actualSlot (utxoHistory w)
         , submissions =
             Delta.apply (Sbm.rollBackward actualSlot) (submissions w)
+        , txHistory =
+            TxHistory.rollBackward
+                resolveSlot
+                (Read.slotFromChainPoint targetPoint)
```

Comment:
```suggestion
                actualSlot
```

We're rolling backward to the `actualSlot` here, which is computed in the `where` clause. 🤓